src/sentry/search/events/builder/metrics.py

```diff
@@ -1307,6 +1323,36 @@ def get_default_value(meta_type: str) -> Any:
         else:
             return None
 
+    def _get_metric_prefix(
+        self, snql_function: fields.MetricsFunction, column: str | None
+    ) -> str | None:
+        if self.use_metrics_layer or self.use_case_id not in {
+            UseCaseID.SPANS,
+            UseCaseID.TRANSACTIONS,
+        }:
+            return None
+
+        prefix_to_function_map = {
+            "d": snql_function.snql_distribution,
+            "s": snql_function.snql_set,
+            "c": snql_function.snql_counter,
+            "g": snql_function.snql_gauge,
+        }
+
+        metrics_map = {
+            UseCaseID.SPANS: constants.SPAN_METRICS_MAP,
+            UseCaseID.TRANSACTIONS: constants.METRICS_MAP,
+        }
+
+        prefix = None
+        primary_metric = metrics_map[self.use_case_id].get(column)
+        prefix = primary_metric.split(":")[0] if primary_metric else None
+        if prefix is not None and prefix_to_function_map.get(prefix) is None:
```

Comment:
nit: let's do so any unknown prefixes get default treatment?
```suggestion
        if prefix is not in prefix_to_function_map and prefix_to_function_map.get(prefix) is None:
```